src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinMapStorage_threadedIO.java

```diff
@@ -0,0 +1,45 @@
+package com.mitchej123.hodgepodge.mixins.early.minecraft;
+
+import java.io.File;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.WorldSavedData;
+import net.minecraft.world.storage.ISaveHandler;
+import net.minecraft.world.storage.MapStorage;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+import com.mitchej123.hodgepodge.config.TweaksConfig;
+import com.mitchej123.hodgepodge.core.HodgepodgeCore;
+
+@Mixin(MapStorage.class)
+public class MixinMapStorage_threadedIO {
+
+    @Shadow
+    private ISaveHandler saveHandler;
+
+    /**
+     * @author mitchej123
+     * @reason Async saving
+     */
+    @Overwrite
+    private void saveData(WorldSavedData data) {
+        if (this.saveHandler == null) return;
+
+        if (!TweaksConfig.saveMineshaftData && data.mapName.equals("Mineshaft")) return;
+
+        File file = this.saveHandler.getMapFileFromName(data.mapName);
+        if (file == null) return;
+
+        NBTTagCompound tag = new NBTTagCompound();
+        data.writeToNBT(tag);
+        NBTTagCompound parentTag = new NBTTagCompound();
+        parentTag.setTag("data", tag);
+
+        HodgepodgeCore.saveWorldData(file, parentTag);
```

Comment:
This is a hard no. These are not moved to threaded IO by forge for a reason. A lot of these WorldSavedData (vanilla and modded) keeps a NBTTagCompound as its runtime data structure, and on writeToNBT() call it just sets it as a subtag without making a copy. Doing async save of these tags will eventually lead to ConcurrentModificationExceptions, e.g. https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12802

Reply:
For clarity: the linked issue is a demonstration of how CME could be triggered by async world save, not actual past incidents. Since WorldSavedData has always been saved synchronously on main thread, it is impossible to have a existing example.

Reply:
I'll add an option to do a copy here then, it should still be faster than doing the save on the main thread. 

I didn't realize people did terrible in writeNbt implementations like that 🤦   I assumed since the tag was written and only used here it would be safe, but reusing something that modified it would indeed cme...

Reply:
Added an interface `SafeWriteNBT` and added it to all classes that seem to safely write the NBT.  If the interface is not present it'll do a .copy().   Took around ~500-1700ms for a ~22mb Mineshaft.dat.  Was significantly faster on normal sized files.

On a related note - I do have ambitions to splitting that up by region as well as some other similar classes (VillageCollection comes to mind), but that'd be a separate PR.